relay-server/src/services/processor/profile_chunk.rs

```diff
@@ -16,7 +16,20 @@ use {
 
 /// Removes profile chunks from the envelope if the feature is not enabled.
 pub fn filter<G>(state: &mut ProcessEnvelopeState<G>) {
-    let continuous_profiling_enabled = state.project_info.has_feature(Feature::ContinuousProfiling);
+    // if:
+    // (not "organizations:continuous-profiling-beta-ingest" && "organizations:continuous-profiling")
+    // ||
+    // ("organizations:continuous-profiling-beta-ingest" && "organizations:continuous-profiling-beta")
+    let continuous_profiling_enabled = (!state
+        .project_info
+        .has_feature(Feature::ContinuousProfilingBetaIngest)
+        && state.project_info.has_feature(Feature::ContinuousProfiling))
+        || (state
+            .project_info
+            .has_feature(Feature::ContinuousProfilingBetaIngest)
+            && state
+                .project_info
+                .has_feature(Feature::ContinuousProfilingBeta));
```

Comment:
Something like this is probably easier to read
```suggestion
    let continuous_profiling_enabled = if state.project_info.has_feature(Feature::ContinuousProfilingBetaIngest) {
        state.project_info.has_feature(Feature::ContinuousProfilingBeta)
    } else {
        state.project_info.has_feature(Feature::ContinuousProfiling)
    };
```